sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -1736,6 +1737,40 @@ class CollationSQLExpressionsSuite
     })
   }
 
+  test("Support Mode.eval(buffer) with complex types") {
+    case class UTF8StringModeTestCase[R](
+      collationId: String,
+      bufferValues: Map[InternalRow, Long],
+      result: R)
```

Comment:
4 spaces, see https://github.com/databricks/scala-style-guide?tab=readme-ov-file#spacing-and-indentation